lib/template-loader.php

```diff
@@ -133,12 +133,32 @@ function gutenberg_resolve_template( $template_type, $template_hierarchy = array
 			'orderby'        => 'post_name__in',
 			'posts_per_page' => -1,
 			'no_found_rows'  => true,
+			// Templates that either:
+			//
+			// - the active theme has declared (stylesheet)
+			// - its parent theme (template) has declared but there's not an equivalent in the active theme
+			//
 			'tax_query'      => array(
+				'relation' => 'OR',
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'slug',
 					'terms'    => wp_get_theme()->get_stylesheet(),
 				),
+				array(
```

Comment:
Depending on how complex this PR gets it may make sense to split into two PR's: One that adds child theme support for FSE and another that adds it to Global Styles.